src/components/screens/vacation-requests/vacation-requests-table/vacation-requests-table-toolbar/toolbar-form/toolbar-form.tsx

```diff
@@ -0,0 +1,161 @@
+import { Box, Divider, Grid } from "@mui/material";
+import {
+  VacationRequest,
+  VacationRequestStatus,
+  VacationType
+} from "../../../../../../generated/client";
+import { Dispatch, SetStateAction, useEffect, useState } from "react";
+import { DateTime } from "luxon";
+import { DataGridRow, VacationData, ToolbarFormModes } from "../../../../../../types";
+import CreateVacationRequest from "../../../create-vacation-request";
+import { GridRowId } from "@mui/x-data-grid";
+import UpdateVacationRequest from "../../../update-vacation-request";
+import FormFields from "./form-fields";
+import { getVacationDataFromRow } from "./get-vacation-data-from-row";
+import { determineToolbarFormMode } from "../../../../../../utils/toolbar-utils";
+
+/**
+ * Component properties
+ */
+interface Props {
+  formOpen: boolean;
+  setFormOpen: Dispatch<SetStateAction<boolean>>;
+  vacationRequests: VacationRequest[];
+  vacationRequestStatuses: VacationRequestStatus[];
+  setVacationRequests: Dispatch<SetStateAction<VacationRequest[]>>;
+  setVacationRequestStatuses: Dispatch<SetStateAction<VacationRequestStatus[]>>;
+  selectedRowIds: GridRowId[] | undefined;
+  rows: DataGridRow[] | undefined;
+  toolbarFormMode: ToolbarFormModes;
+  setToolbarFormMode: Dispatch<SetStateAction<ToolbarFormModes>>;
+}
+
+/**
+ * Table form component
+ *
+ * @param props component properties
+ */
+const TableForm = ({
+  formOpen,
+  setFormOpen,
+  vacationRequestStatuses,
+  setVacationRequestStatuses,
+  vacationRequests,
+  setVacationRequests,
+  selectedRowIds,
+  rows,
+  toolbarFormMode,
+  setToolbarFormMode
+}: Props) => {
+  const dateTimeNow = DateTime.now();
+  const [initialStartDate, setInitialStartDate] = useState<DateTime | undefined>(dateTimeNow);
+  const [initialEndDate, setInitialEndDate] = useState<DateTime | undefined>(dateTimeNow);
+  const [vacationData, setVacationData] = useState<VacationData>({
+    startDate: dateTimeNow,
+    endDate: dateTimeNow,
+    type: VacationType.VACATION,
+    message: "",
+    days: 1
+  });
+  const [selectedVacationRequestId, setSelectedVacationRequestId] = useState<string | undefined>(
```

Comment:
Does this need to be typed | undefined? Could it be initialized with an empty string?

Reply:
Initialized with empty string.